server/routes/handlers/helpers/dates.ts

```diff
@@ -1,32 +1,53 @@
-import { getTimezoneOffset } from 'date-fns-tz'
-import { isValid } from 'date-fns'
+import { DateTime, Settings } from 'luxon'
 import { DatePartsParsed, ObjectMap } from '../../../@types'
 import logger from '../../../../logger'
 import { isDefined } from './index'
 
-const getDaylightSavingOffset = (d: Date) => getTimezoneOffset('Europe/London', d) / (1000 * 60 * 60)
+const timeZone = 'Europe/London'
+Settings.throwOnInvalid = true
 
-export const convertGmtDatePartsToUtc = ({ year, month, day, hour, minute }: ObjectMap<string>): string | undefined => {
+interface Options {
+  dateMustBeInPast?: boolean
+  dateMustBeInFuture?: boolean
+}
+
+export const convertGmtDatePartsToUtc = (
+  { year, month, day, hour, minute }: ObjectMap<string>,
+  options?: Options
+): string | undefined => {
   try {
     const includeTime = isDefined(hour) && isDefined(day)
     const parts = [year, month, day, ...(includeTime ? [hour, minute] : [])]
-    if (parts.find(part => !Number.isInteger(parseInt(part, 10)))) {
+    const hasInvalidPart = parts.some(part => !Number.isInteger(parseInt(part, 10)))
+    if (hasInvalidPart) {
       return undefined
     }
     const [y, m, d, h, min] = parts.map(part => {
       return parseInt(part, 10)
     })
-    let date
-    if (includeTime) {
-      date = new Date(Date.UTC(y, m - 1, d, h, min))
-      date.setHours(date.getHours() - getDaylightSavingOffset(date))
-    } else {
-      date = new Date(Date.UTC(y, m - 1, d))
+    const date = DateTime.fromObject(
+      {
+        year: y,
+        month: m,
+        day: d,
+        hour: h,
+        minute: min,
+      },
+      { zone: timeZone }
+    )
+    if (options) {
+      const now = DateTime.now()
+      if (options.dateMustBeInPast === true && now < date) {
+        return undefined
+      }
+      if (options.dateMustBeInFuture === true && now > date) {
+        return undefined
+      }
     }
-    if (!isValid(date)) {
-      return undefined
+    if (includeTime) {
+      return date.setZone('utc').toString()
     }
-    return includeTime ? date.toISOString() : date.toISOString().substring(0, 10)
+    return date.toISODate()
   } catch (err) {
```

Comment:
That's a massive try/catch block isn't it? Perhaps the code can be refactored to make it clearer which exception is meant to be caught?